src/components/widgets/sales-average-purchase-size/sales-average-purchase-size.directive.coffee

```diff
@@ -0,0 +1,88 @@
+#
+# Component generated by Impac! Widget Generator!
+#
+module = angular.module('impac.components.widgets.sales-average-purchase-size', [])
+module.controller('WidgetSalesAveragePurchaseSizeCtrl', ($scope, $q, $filter, ImpacWidgetsSvc, ImpacAssets, HighchartsFactory, ImpacUtilities) ->
+
+  w = $scope.widget
+  $scope.isChartDisplayed = true
+
+  # Define settings
+  # --------------------------------------
+  $scope.orgDeferred = $q.defer();
+  settingsPromises = [$scope.orgDeferred.promise]
+
+  # Time management
+  todayUTC = moment().startOf('day').add(moment().utcOffset(), 'minutes')
+
+  # Timestamps stored in the back-end are in UTC => the filter on the date must be UTC too
+  dateFilter = (timestamp) ->
+    pickedDate = moment.utc(timestamp)
+    if pickedDate <= todayUTC then "lte #{pickedDate.format('YYYY-MM-DD')}" else pickedDate.format('YYYY-MM-DD')
+
+  # Unique identifier for the chart object in the DOM
+  $scope.chartId = ->
+    "averagePurchaseSizeChart-#{w.id}"
+
+  # # == Sub-Components - Needed? =============================================================
+  $scope.chartDeferred = $q.defer()
+
+  # == Chart Events Callbacks =====================================================================
+  # Sets the transactions list resources type and displays it
+  onClickBar = (event) ->
+    $scope.isChartDisplayed = false
+
+  # == Directive Events Callbacks =====================================================================
+  $scope.onButtonBack = () ->
+    $scope.isChartDisplayed = true
+
+  # Widget specific methods
+  # --------------------------------------
+  w.initContext = ->
+    $scope.isDataFound = w.content?
+
+  w.format = ->
+    # Instantiate and render chart
+    options =
+      chartType: 'line'
+      chartOnClickCallbacks: []
+      currency: w.metadata.currency
+      showToday: true
+      showLegend: true
+
+    $scope.chart = new HighchartsFactory($scope.chartId(), w.content.chart, options)
+
+    $scope.chart.formatters = ->
+        currency = @options.currency
+        xAxisLabels =
+          labels:
+            formatter: ->
+              if (this.chart.rangeSelector.options.selected >= 3)
+                moment.utc(this.value).format('MMM YYYY')
+              else
+                moment.utc(this.value).format('DD MMM')
+        yAxisLabels =
+          labels:
+            formatter: ->
+              $filter('mnoCurrency')(this.value, currency, false)
+        xAxis: angular.merge([w.content.chart.xAxis[0]], [xAxisLabels])
+        yAxis: angular.merge([w.content.chart.yAxis[0]], [yAxisLabels])
```

Comment:
The yAxisLabels are the same as default (right?), for overriding the formatters method, I suggest leveraging the formatters method's returned defaults by savings them to a variable in the outer scope.

Also, rather than having to merge the new v2 bolt layouts response options (xAxis, yAxis, tooltip etc) for every single custom widget template, the highcharts factory should be extended to use these by default.

 Then you could do this sort of thing:
```coffeescript
defaultFormatters = $scope.chart.formatters()
$scope.chart.formatters = ->
  defaultFormatters.labels.formatter = -> 
    formatString = if this.chart.rangeSelector.options.selected >= 3 then 'MMM YYY' else 'DD MMM'
    moment.utc(this.value).format(formatString)
  defaultFormatters.rangeSelector = { selected: 5 }
  defaultFormatters
```

EDIT: so I can see now that you have extended the highcharts formatter, so would simply be saving the `defaultFormatters` be enough to not have to do the `angular.merge([w.content.chart.xAxis[0] .. )` ? Seems like you are giving the formatter the w.content.chart.xAxis in initialization, but then having to re-provide it with it to merge in the custom opts.

Reply:
@xaun I am not completely sure about this as there seem to be slight differences in the overridden formatters

Reply:
@MarcoBgn Im not completely across the intricacies of your customisations, so my solution isn't comprehensive for your use-case. I am just showing you a way you can store the existing defaults, then modify the object and reassign the formatters method. 

The main thing is, this method is massive in the widget controller, it redefines **some** values that are already defined in the default, and looks very similar across all of the new widgets. So for my review, I request that we think of a solution to improve this. If this is not in the scope of this body of work, or it needs to be quick & dirty merged or something, that's another story :)